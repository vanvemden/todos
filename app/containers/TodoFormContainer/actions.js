/*
 *
 * TodoFormContainer actions
 *
 */

import {
  SUBMIT_TODO,
  SUBMIT_TODO_SUCCEEDED,
  SUBMIT_TODO_FAILED,
  SUBMIT_TODO_CANCELLED,
} from './constants';

export function submitTodo() {
  return {
    type: SUBMIT_TODO,
  };
}

export function submitTodoSucceeded(todo) {
  return {
    type: SUBMIT_TODO_SUCCEEDED,
    todo,
  };
}

export function submitTodoFailed(message) {
  return {
    type: SUBMIT_TODO_FAILED,
    message,
  };
}

export function submitTodoCancelled() {
  return {
    type: SUBMIT_TODO_CANCELLED,
  };
}
