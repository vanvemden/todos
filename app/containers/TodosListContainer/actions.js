/*
 *
 * TodosListContainer actions
 *
 */

import {
  REQUEST_TODOS,
  REQUEST_TODOS_SUCCEEDED,
  REQUEST_TODOS_FAILED,
  TOGGLE_TODO,
  TOGGLE_TODO_SUCCEEDED,
  TOGGLE_TODO_FAILED,
} from './constants';

export function requestTodos() {
  return {
    type: REQUEST_TODOS,
  };
}

export function requestTodosSucceeded(todos) {
  return {
    type: REQUEST_TODOS_SUCCEEDED,
    todos,
  };
}

export function requestTodosFailed(message) {
  return {
    type: REQUEST_TODOS_FAILED,
    message,
  };
}

export function toggleTodo(id) {
  return {
    type: TOGGLE_TODO,
    id,
  };
}

export function toggleTodoSucceeded(id) {
  return {
    type: TOGGLE_TODO_SUCCEEDED,
    id,
  };
}

export function toggleTodoFailed(message) {
  return {
    type: TOGGLE_TODO_FAILED,
    message,
  };
}
