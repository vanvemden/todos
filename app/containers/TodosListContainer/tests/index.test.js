/**
 *
 * Tests for TodosListContainer
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

// import 'jest-dom/extend-expect'; // add some helpful assertions
import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';

import { TodosListContainer, mapDispatchToProps } from '../index';
import { requestTodos, toggleTodo } from '../actions';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<TodosListContainer />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <TodosListContainer
            todos={[]}
            onRequestTodos={() => {}}
            onToggleTodo={() => {}}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <TodosListContainer
            todos={[]}
            onRequestTodos={() => {}}
            onToggleTodo={() => {}}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('Should fetch the todos on mount', () => {
    const submitSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <TodosListContainer
            todos={[]}
            onRequestTodos={submitSpy}
            onToggleTodo={() => {}}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onRequestTodos', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onRequestTodos).toBeDefined();
      });

      it('should dispatch requestTodos when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onRequestTodos();
        expect(dispatch).toHaveBeenCalledWith(requestTodos());
      });
    });

    describe('onToggleTodo', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onToggleTodo).toBeDefined();
      });

      it('should dispatch toggleTodo when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const id = '12345';
        result.onToggleTodo(id);
        expect(dispatch).toHaveBeenCalledWith(toggleTodo(id));
      });
    });
  });
});
