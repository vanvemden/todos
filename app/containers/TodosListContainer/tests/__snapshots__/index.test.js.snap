// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodosListContainer /> Should render and match the snapshot 1`] = `
<div>
  <h1>
    <span>
      to-do list
    </span>
  </h1>
  <div
    class="TodoList__StyledWrapper-sc-11d6ln9-0 dWPHBJ"
  >
    <ul
      class="TodoList__StyledUl-sc-11d6ln9-1 kTWTrU"
    />
  </div>
</div>
`;
